src/main/java/it/gov/pagopa/pu/migration/service/file/ErrorArchiverService.java

```diff
@@ -0,0 +1,117 @@
+package it.gov.pagopa.pu.migration.service.file;
+
+import it.gov.pagopa.pu.migration.dto.ErrorFileDTO;
+import it.gov.pagopa.pu.migration.utils.Utilities;
+import it.gov.pagopa.pu.migration.wf.exception.NotRetryableActivityException;
+import it.gov.pagopa.pu.p4paprocessexecutions.dto.generated.IngestionFlowFile;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.CollectionUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * A base service for archiving error files.
+ * This class provides common functionality for writing error data to CSV files and archiving them into ZIP archives.
+ * It is designed to be extended by specific error archiver services for different types of error DTOs.
+ *
+ * @param <T> The type of the error DTO.
+ */
+@Slf4j
+public abstract class ErrorArchiverService<T extends ErrorFileDTO> {
+
+    private static final String ERRORFILE_PREFIX = "ERROR-";
+
+    private final Path sharedDirectoryPath;
+    private final String errorFolder;
+    private final FileArchiverService fileArchiverService;
+    private final CsvService csvService;
+
+    protected ErrorArchiverService(
+            String sharedFolder,
+            String errorFolder,
+            FileArchiverService fileArchiverService,
+            CsvService csvService
+    ) {
+        this.sharedDirectoryPath = Path.of(sharedFolder);
+        this.errorFolder = errorFolder;
+        this.fileArchiverService = fileArchiverService;
+        this.csvService = csvService;
+    }
+
+
+    protected abstract List<String[]> getHeaders();
+
+    /**
+     * Writes error data into a CSV file.
+     *
+     * @param workingDirectory  The directory where the file should be created.
+     * @param ingestionFlowFile The metadata of the ingestion file.
+     * @param errorList         The list of errors to write.
+     */
+    public void writeErrors(Path workingDirectory, IngestionFlowFile ingestionFlowFile, List<T> errorList) {
```

Comment:
why IngestionFlowFile parameter? here you will not handle IngestionFlowFile entities...